File: dcget/src/hub/LockHandler.java
package hub;

class LockHandler implements IHubHandler {

    private HubConnection mgr;

    public LockHandler(HubConnection mgr) {
        this.mgr = mgr;
    }

    public void handleHubCommand(byte[] data) throws Exception {
        String d = new String(data);
        if (!d.startsWith("$Lock ")) {
            return;
        }
        String[] s = d.split(" ");
        if (s.length < 2) {
            return;
        }
        mgr.onHubConnected(s[1]);
    }

}

File: dcget++/LockHandler.cpp
#include "LockHandler.h"

void LockHandler::handleHubCommand(const std::string& data) {
    if (data.find("$Lock") != 0)
        return;
    std::vector<std::string> s = StringUtils::split(data, ' ');
    if (s.size() < 2)
        throw Exception("lock not found");
    mgr->onHubConnected(s[1]);
}

